### PR TITLE
Compatible with proto.MessageV1

### PR DESCRIPTION
**Pull request by 2zturn**

### CODE REVIEW

**Comment by wu-sheng** [comment]
Could you be more clear about what is this about? What V1 message?

File: plugins/server/grpc/server_codec.go [hunk 1]
```diff
@@ -41,28 +42,42 @@ func NewOriginalData(data []byte) *OriginalData {
...
 		return nil
 	}
-	return proto.Unmarshal(data, vv)
+
+	vv, ok := v.(proto.Message)
+	if ok {
+		return proto.Unmarshal(data, vv)
+	}
+
+	err := proto.Unmarshal(data, pbv1.MessageV2(v))
```
Comment (mrproliu) [inline plugins/server/grpc/server_codec.go:75]: I think It needs to use `MessageV1` method?

File: plugins/server/grpc/server_codec.go [hunk 2]
```diff
@@ -41,28 +42,42 @@ func NewOriginalData(data []byte) *OriginalData {
...
 		return original.Content, nil
 	}
-	return proto.Marshal(vv)
+
+	vv, ok := v.(proto.Message)
+	if ok {
+		return proto.Marshal(vv)
+	}
+
+	b, err := pbv1.Marshal(pbv1.MessageV1(v))
```
Comment (mrproliu) [inline plugins/server/grpc/server_codec.go:55]: Please use the cast to check again.

**Review by mrproliu — COMMENTED** [review]
Thanks for your contribution. Please fix the comment and add the changes to the [CHANGES.md](https://github.com/apache/skywalking-satellite/blob/main/CHANGES.md).

File: plugins/server/grpc/server_codec.go (hunk 1 above)
Comment (2zturn) [reply plugins/server/grpc/server_codec.go:75]: in my case, I use MessageV1 at first, but it doesn't work,  when change it  to MessageV2, it's work, so ...

**Comment by mrproliu** [comment]
@2zturn Could you help when we need to use the V1 message? When will `V1` be needed?

**Comment by 2zturn** [comment]
when it works with: go.opentelemetry.io/collector/model/otlpgrpc,  this package will return the MessageV1, then stream it, the origin marshal is failed

**Comment by wu-sheng** [comment]
> when it works with: go.opentelemetry.io/collector/model/otlpgrpc, this package will return the MessageV1, then stream it, the origin marshal is failed

I am a little confusing. What gRPC service do you use to forward data to Satellite?

**Comment by 2zturn** [comment]
> > when it works with: go.opentelemetry.io/collector/model/otlpgrpc, this package will return the MessageV1, then stream it, the origin marshal is failed
> 
> I am a little confusing. What gRPC service do you use to forward data to Satellite?

I pull data from Satellite not forward to it

**Comment by wu-sheng** [comment]
`Pull from satellite` to `where`?

**Comment by JaredTan95** [comment]
> `Pull from satellite` to `where`? Which format is compatible with OTEL collector? 

convert skywalking segment to otlp tracing

**Comment by wu-sheng** [comment]
> > `Pull from satellite` to `where`? Which format is compatible with OTEL collector?
> 
> convert skywalking segment to otlp tracing

But this code is adding to our gRPC server, shouldn't this be added into OTEL collector? I think I need more information up to your proposal

**Comment by 2zturn** [comment]
ok, i will try to add into OTEL collector, thx for your all

**Pull request closed without merging**